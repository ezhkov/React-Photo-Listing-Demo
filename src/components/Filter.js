import React, { Component } from 'react';
import {Dropdown, DropdownToggle, DropdownMenu, DropdownItem } from 'mdbreact';


export class Filter extends Component {

    state = {};


    render() {
        return (
            <Dropdown className="ml-3">
                <DropdownToggle caret color="primary ml-0">
                    Material dropdown
                </DropdownToggle>
                <DropdownMenu>
                    <DropdownItem href="#">Action</DropdownItem>
                    <DropdownItem href="#">Another Action</DropdownItem>
                    <DropdownItem href="#">Something else here</DropdownItem>
                    <DropdownItem href="#">Something else here</DropdownItem>
                </DropdownMenu>
            </Dropdown>
        );
    }

}