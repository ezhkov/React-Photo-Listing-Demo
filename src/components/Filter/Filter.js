import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ClickOutside from 'react-click-outside';
import { Dropdown, DropdownToggle, DropdownMenu } from 'mdbreact';
import DropdownItem from '../DropdownItem/DropdownItem';

class Filter extends Component {
  constructor(...args) {
    super(...args);
    const { activeFilter } = this.props;
    this.state = {
      activeFilter,
      showDropdown: false,
    };
  }

  onFilterClick = (item) => {
    const { getFilterItemValue } = this.props;
    this.setState({ activeFilter: item.label, showDropdown: false });
    getFilterItemValue(item);
  };

  hide = () => {
    this.setState({ showDropdown: false });
  };

  toggle = () => {
    const { showDropdown } = this.state;
    this.setState({ showDropdown: !showDropdown });
  };

  renderListItems = (filters) => {
    const { activeFilter } = this.state;
    return filters.map(item => (
      <DropdownItem
        key={item.label}
        activeFilter={activeFilter}
        filterType={item.label}
        hasPadding={item.hasPadding}
        onFilterClick={() => this.onFilterClick(item)}
      />
    ));
  };

  render() {
    const { activeFilter, showDropdown } = this.state;
    const { filters, buttonColor } = this.props;

    return (
      <ClickOutside className="clickOutSide" onClickOutside={this.hide}>
        <Dropdown className="ml-3">
          <DropdownToggle caret className="filter__toggler" color={`${buttonColor} ml-0`} onClick={this.toggle}>
            {activeFilter}
          </DropdownToggle>
          {showDropdown && (
          <DropdownMenu className="filter-dropDownMenu" style={{ display: 'block' }}>
            {this.renderListItems(filters)}
          </DropdownMenu>
          )}
        </Dropdown>
      </ClickOutside>
    );
  }
}


Filter.propTypes = {
  activeFilter: PropTypes.string,
  filters: PropTypes.arrayOf(PropTypes.object),
  buttonColor: PropTypes.string,
  getFilterItemValue: PropTypes.func,
};
Filter.defaultProps = {
  activeFilter: '',
  buttonColor: 'primary',
  filters: [],
  getFilterItemValue: () => {},
};
export default Filter;
