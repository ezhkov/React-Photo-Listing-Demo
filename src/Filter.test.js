import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Filter from './components/Filter';


Enzyme.configure({ adapter: new Adapter() });

describe('Test of component of Filter.js', () => {
  it('Initial filter\'s state showDropdown ', () => { 
    const filter = shallow(<Filter />)
    expect(filter.state().showDropdown).toEqual(false);
  });

  it('Initial filter\'s state showDropdown ', () => { 
    const filter = shallow(<Filter />)
    //expect(filter.find('t').at(1).props().caret).toBe(true);
    //expect(filter).toMatchSnapshot()
    //const tree = create(<Filter />).toJSON();
    //filter.find('t').at(1).simulate('click');
    //const tree = renderer.create(<Filter />).toJSON();

    //expect(tree).toMatchSnapshot()
    //expect(filter.find('div')).toBe(true);
    //expect(filter.exists('div')).toEqual(true);
    //console.log(filter.debug());
    //console.log(filter.state());
    //console.log('111')

  });
});