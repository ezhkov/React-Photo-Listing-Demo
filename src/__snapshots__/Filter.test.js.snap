// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of component of Filter.js Initial filter's state showDropdown  1`] = `
<div>
  <div
    className="dropdown ml-3"
    onKeyDown={[Function]}
  >
    <button
      aria-expanded={false}
      aria-haspopup={true}
      className="btn btn-primary ml-0 Ripple-parent dropdown-toggle"
      onClick={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
      type="button"
    >
      <span
        className="sr-only"
      >
        Toggle Dropdown
      </span>
      <div
        className="Ripple "
        style={
          Object {
            "height": "0px",
            "left": "0px",
            "top": "0px",
            "width": "0px",
          }
        }
      />
    </button>
  </div>
</div>
`;
